src/components/AdminPanelLayout/UsersPane.tsx

```diff
@@ -1,21 +1,33 @@
-import {Table} from 'react-bootstrap';
+import {Button, Table} from 'react-bootstrap';
 import React, {useState} from 'react';
 import {User} from '../../types/User';
-import {getUserList} from '../../api';
+import {changeUserRole, getUserList} from '../../api';
 import {useMount} from '../../hooks/useMount';
 import {HorizontallyCenteredSpinner} from '../util/HorizonallyCenteredSpinner';
 import {Layout} from '../../pages/Layout/Layout';
-
+import UserListItem, {ChangeUserRoleRequestSchema} from './UserListItem';
 
 const UsersPane = () => {
   const [users, setUsers] = useState();
   const [loading, setLoading] = useState(true);
+  const [updateRequests, setUpdateRequests] = useState<ChangeUserRoleRequestSchema[]>([]);
 
   useMount(async () => {
     setUsers(await getUserList());
     setLoading(false);
   });
 
+  const addUserToUpdate = (request: ChangeUserRoleRequestSchema) => {
+    const listToUpdate = updateRequests;
+    listToUpdate.push(request);
+    setUpdateRequests(listToUpdate);
```

Comment:
You can avoid the temporary variables with the spread operator.
```suggestion
    setUpdateRequests([...updateRequests, request]);
```